### DISCUSSION TITLE
Issue using Livewire along with Alpine JS x-if directive

### DISCUSSION THREAD

**Discussion by intrepidws** [livewire/livewire#5110] [1. Help] [answered]
I'm running into trouble when using the Alpine JS `x-if` directive within a Livewire component. I also get the problem when doing the reverse: using a Livewire component within a `x-if` directive.

The simplified code looks something like:

```
<template x-if="isMember">
    <div class="card-body">
        <p><em>Please specify the postal code of your member facility.</em></p>

        <livewire:member-lookup />
    </div>
</template>
```

I don't get any errors (either in Laravel or in Chrome DevTools) but the component just doesn't work. If I use `x-show` instead of `x-if`, it seems to work fine. This code works:

```
<div x-show="isMember">
    <div class="card-body">
        <p><em>Please specify the postal code of your member facility.</em></p>

        <livewire:member-lookup />
    </div>
</div>
```

Is it just not possible to use a Livewire component along with the Alpine `x-if` directive?

**Comment by joshhanley** [comment, marked as answer]
@intrepidws nah, you can't use Livewire directives with Alpine's `x-if`.

The reason is Livewire only processes the directives (`wire:model`, `wire:click`, etc) whenever a Livewire request happens. But as the elements don't exist on the page when `x-if` is `false`, then Livewire won't process them.

Plus any `<livewire:... />` tag is actually handled and processed by blade, so even if Livewire could react when `x-if` was set to `true`, the Livewire component tags still wouldn't be processed.

Due to this, you'd be better of using `x-show` so the Livewire component and directives already exist on the page, or if you don't want them in the DOM, then you can use Livewire to control it by wrapping the Livewire component in a blade `@if` and toggling the visibility in a Livewire property.

Hope this helps!

**Comment by intrepidws** [comment]
Thanks for your help @joshhanley that makes a lot of sense the way you've explained it.

**Reply by joshhanley** [reply]
@intrepidws no worries! 🙂
